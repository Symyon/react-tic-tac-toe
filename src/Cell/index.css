.Cell {
    display: flex;
    cursor: pointer;
    background-color: green;
    width: 32%;
    height: 32%;
    justify-content: center;
    align-items: center;
}