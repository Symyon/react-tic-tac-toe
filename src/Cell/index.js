import React, { Component } from 'react';
import nought from './img/o.png';
import cross from './img/x.png';
import './index.css';

class Cell extends Component {
    render() {

        const { mark, onClick, index } = this.props;
        return (
            <div className="Cell"
                onClick={() => onClick(index)}>
                {mark !== -1 && (mark === 0 ?
                    <img src={nought} alt={"O"} style={{ width: '100%', height: '100%' }} /> :
                    <img src={cross} alt={"X"} style={{ width: '100%', height: '100%' }} />)}
            </div>
        );
    }
}

export default Cell;