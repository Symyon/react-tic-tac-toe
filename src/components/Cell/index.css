.Cell {
    display: flex;
    cursor: pointer;
    background-color: rgb(87, 90, 72);
    width: 32%;
    height: 32%;
    justify-content: center;
    align-items: center;
}