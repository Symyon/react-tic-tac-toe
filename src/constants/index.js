export const EMPTY = -1;
export const NOUGHT = 0;
export const CROSS = 1;
export const DRAW = 2;