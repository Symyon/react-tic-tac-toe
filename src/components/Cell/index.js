import React, { Component } from 'react';
import nought from './img/o.png';
import cross from './img/x.png';
import { NOUGHT, EMPTY } from '../../constants';
import './index.css';

class Cell extends Component {
    render() {

        const { mark, onClick, index, markIt } = this.props;
        return (
            <div className="Cell"
                onClick={() => onClick(index)}
                style={{ backgroundColor: markIt ? 'rgb(57, 60, 42)' : 'rgb(87, 90, 72)' }}>
                {mark !== undefined && mark !== EMPTY && (mark === NOUGHT ?
                    <img src={nought} alt={"O"} style={{ width: '100%', height: '100%' }} /> :
                    <img src={cross} alt={"X"} style={{ width: '100%', height: '100%' }} />)}
            </div>
        );
    }
}

export default Cell;