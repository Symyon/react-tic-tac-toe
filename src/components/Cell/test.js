import React from 'react';
import { shallow } from 'enzyme';
import Cell from './';

import { NOUGHT, CROSS, EMPTY } from '../../constants';

describe('Cell', () => {
    it('render correctly without props', () => {
        const component = shallow(<Cell />);
        expect(component).toMatchSnapshot();
    })

    it('render correctly empty', () => {
        const component = shallow(<Cell mark={EMPTY} />);
        expect(component).toMatchSnapshot();
    })

    it('render correctly with cross', () => {
        const component = shallow(<Cell mark={CROSS} />);
        expect(component).toMatchSnapshot();
    })

    it('render correctly with nought', () => {
        const component = shallow(<Cell mark={NOUGHT} />);
        expect(component).toMatchSnapshot();
    })

    it('render correctly when marked as winning', () => {
        const component = shallow(<Cell markIt={true} />);
        expect(component).toMatchSnapshot();
    })

    it('render correctly when marked as not winning', () => {
        const component = shallow(<Cell markIt={false} />);
        expect(component).toMatchSnapshot();
    })
});