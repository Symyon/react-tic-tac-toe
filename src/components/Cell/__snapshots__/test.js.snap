// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cell render correctly empty 1`] = `
<div
  className="Cell"
  onClick={[Function]}
  style={
    Object {
      "backgroundColor": "rgb(87, 90, 72)",
    }
  }
/>
`;

exports[`Cell render correctly when marked as not winning 1`] = `
<div
  className="Cell"
  onClick={[Function]}
  style={
    Object {
      "backgroundColor": "rgb(87, 90, 72)",
    }
  }
/>
`;

exports[`Cell render correctly when marked as winning 1`] = `
<div
  className="Cell"
  onClick={[Function]}
  style={
    Object {
      "backgroundColor": "rgb(57, 60, 42)",
    }
  }
/>
`;

exports[`Cell render correctly with cross 1`] = `
<div
  className="Cell"
  onClick={[Function]}
  style={
    Object {
      "backgroundColor": "rgb(87, 90, 72)",
    }
  }
>
  <img
    alt="X"
    src="x.png"
    style={
      Object {
        "height": "100%",
        "width": "100%",
      }
    }
  />
</div>
`;

exports[`Cell render correctly with nought 1`] = `
<div
  className="Cell"
  onClick={[Function]}
  style={
    Object {
      "backgroundColor": "rgb(87, 90, 72)",
    }
  }
>
  <img
    alt="O"
    src="o.png"
    style={
      Object {
        "height": "100%",
        "width": "100%",
      }
    }
  />
</div>
`;

exports[`Cell render correctly without props 1`] = `
<div
  className="Cell"
  onClick={[Function]}
  style={
    Object {
      "backgroundColor": "rgb(87, 90, 72)",
    }
  }
/>
`;
